### ISSUE TITLE
wav2vec pretrain help

### ISSUE THREAD

**Issue by lixx0105** [pytorch/fairseq#3834]
## ❓ Questions and Help
i'm trying to pretrain custom wav2vec2 model using my own dataset. The dataset size is about 10k hour. The official wav2vec2 base model  was used as parameter initialization. The training loss suddenly drop a lot after a few epoch training and validation loss become higher. 

### Before asking:   
1. search the issues.   
2. search the docs.    

#### What is your question?
1.  Training loss (purple )doesn't look right.   Validation loss (red) become higher
![image](https://user-images.githubusercontent.com/48235864/131344623-3d936b91-9b1d-4b6a-a59c-9214ea532b6c.png)

2. does code perplexity curve look normal?
![image](https://user-images.githubusercontent.com/48235864/131344878-fc8038a8-b8ff-4e81-8c7d-0cf0ecaa338c.png)

3. does gradient curve look normal?
![image](https://user-images.githubusercontent.com/48235864/131344969-da89e534-1cd2-4fd5-9fde-047f78fc4305.png)

@alexeib can you kindly help? Thanks.

#### Code
use the same config as wav2vec2 base model.

   

#### What have you tried?
i tried lower the learning rate and fp32 training instead of fp16, but doesn't help.

#### What's your environment?

 - fairseq Version (e.g., 1.0 or master): master
 - PyTorch Version (e.g., 1.0)1.7.1
 - OS (e.g., Linux): Linnux
 - How you installed fairseq (`pip`, source): source
 - Build command you used (if compiling from source):pip install -e 
 - Python version:3.7
 - CUDA/cuDNN version:11.0
 - GPU models and configuration: 4 V100
 - Any other relevant information:

**Comment by alexeib** [comment]
how about the accuracy?
why do you think the drop in loss is necessarily a bad thing? have you tried training to some convergence and evaluating?

**Comment by lixx0105** [comment]
> how about the accuracy?
purple curve is training and red is validation.
![6C3BC75A-AE31-4231-9DE3-9FCC38B76570](https://user-images.githubusercontent.com/48235864/131425786-a4cebb89-3744-4bee-af5f-b5d4da6c4bd4.png)

> why do you think the drop in loss is necessarily a bad thing? have you tried training to some convergence and evaluating?
@alexeib thanks for your quick response. My observation is that code perplexity drops as well as gnorm jumps very high around 17k update. Also the validation loss become higher. So I guess probably something is wrong.
I stopped training when I observed this and haven’t tried to do ASR finetune. Are you suggesting this is ok and I should continue to train the model? Thanks

**Comment by alexeib** [comment]
i think its worth training to convergence and then seeing if the model has learned anything (e.g. via finetuning). this doesnt look like it has completely collapsed to me

**Comment by lixx0105** [comment]
> i think its worth training to convergence and then seeing if the model has learned anything (e.g. via finetuning). this doesnt look like it has completely collapsed to me

ok. i'll continue to run this pretrain. BTW what is reasonable code perplexity / accuracy range for pretrain?

**Comment by RuABraun** [comment]
@alexeib hi, how important is using the settings of the voxlib config like `extractor_mode: layer_norm` and `layer_norm_first: true` when one is training on larger datasets, for example ~5k hours? I'm finding that the loss converges to a significantly higher value when I use larger datasets.

**Comment by alexeib** [comment]
extractor_mode: layer_norm is much more stable and typically has similar performance to default (just make sure you set feature_grad_mult to 1.0 and task.normalize=true)

layer_norm_first allows you to train beyond 500k updates without crashing in fp16 mode. by itself it is not as accurate as post layer norm, but when you train for longer you outperform post layer norm models. for this to be effective you you need to significantly increase the learning rate as compared to post layer norm model (by 20-30x)

**Comment by RuABraun** [comment]
Thank you for explaining!
